src/pages/fi-opening-balances/forms/FiOpeningBalancesForm.js

```diff
@@ -0,0 +1,212 @@
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { useForm } from 'src/hooks/form'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+export default function FiOpeningBalancesForms({ labels, maxAccess, recordId }) {
+  const [editMode, setEditMode] = useState(!!recordId)
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.FiOpeningBalance.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      accountId: null,
+      fiscalYear: null,
+      amount: null,
+      baseAmount: null,
+      currencyId: null,
+      plantId: null
+    },
+    maxAccess: maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      accountId: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      amount: yup.string().required(' '),
+      baseAmount: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+
+      const response = await postRequest({
+        extension: FinancialRepository.FiOpeningBalance.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        toast.success('Record Added Successfully')
+        formik.setValues({
+          ...obj,
+          recordId: response.recordId 
+        })
+      } else toast.success('Record Edited Successfully')
+      setEditMode(true)
+
+      invalidate()
+    }
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+
+          const res = await getRequest({
+            extension: FinancialRepository.FiOpeningBalance.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues(res.record)
+        }
+      } catch (e) { }
+    })()
+  }, [])
+
+  return (
+    <FormShell resourceId={ResourceIds.FiOpeningBalances} form={formik} maxAccess={maxAccess} editMode={editMode}>
+      <Grid container spacing={4}>
+        <Grid item xs={12}>
+          <ResourceComboBox
+            endpointId={SystemRepository.FiscalYears.qry}
+            name='fiscalYear'
+            label={labels.fiscalYear}
+            valueField='fiscalYear'
+            displayField='fiscalYear'
+            values={formik.values}
+            required
+            readOnly={editMode}
+            maxAccess={maxAccess}
+            onChange={(event, newValue) => {
+              formik.setFieldValue('fiscalYear', newValue?.fiscalYear || null)
+            }}
+            error={formik.touched.fiscalYear && Boolean(formik.errors.fiscalYear)}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <ResourceLookup
+            endpointId={FinancialRepository.Account.snapshot}
+            parameters={{
+              _countryId: formik.values.accountId,
```

Comment:
is account snapshot asking for _countryId and _stateId ? 
I don't think so. Please send the necessary parameters to any API request